operator/pkg/util/kyma_util.go

```diff
@@ -26,20 +33,38 @@ func SetComponentCRLabels(unstructuredCompCR *unstructured.Unstructured, compone
 	unstructuredCompCR.SetLabels(labelMap)
 }
 
-func CopyComponentSettingsToUnstructuredFromResource(resource *unstructured.Unstructured, component operatorv1alpha1.ComponentType) {
-	if len(component.CustomStates) > 0 {
-		resource.Object["spec"].(map[string]interface{})["customStates"] = component.CustomStates
+func CopySettingsToUnstructuredFromResource(resource *unstructured.Unstructured, settings []operatorv1alpha1.Settings) {
+	if len(settings) > 0 {
+		resource.Object["spec"].(map[string]interface{})["customStates"] = settings
```

Comment:
we need to rethink this logic. 
settings should ideally be used for spec override
and customStates for state handling of non-native resource

Reply:
For sure, I think we need to extract this logic into a separate key and then align on 2 keys for state checks and settings.